www/js/lib/uiUtil.js

```diff
@@ -183,15 +183,27 @@ function slideAway (e) {
  * @param {String} closeMessageLabel  The text to display on the close alert message button (optional, Default = "Okay")
  * @param {String} hideOptionLabel  The text to display on the hide option button (optional, Default = "Don't ask again")
  * @param {Boolean} displayHideOption If true, option to permanently hide the modal will be shown (currently only implemented for hideExternalLinkWarning)
+ * @param {Object} archiveMetadata If warning message for unknown ZIM file, contains the information to show user to help them confirm validity of ZIM file
+ * @param {String} archiveMetadata.name The name of the ZIM file
+ * @param {String} archiveMetadata.creator The creator of the ZIM file
+ * @param {String} archiveMetadata.publisher The publisher of the ZIM file
+ * @param {String} archiveMetadata.scraper The scraper used to create the ZIM file
  * @returns {Promise<Boolean>} A promise which resolves to true if the user clicked Confirm, false if the user clicked Cancel/Okay, backdrop or the cross(x) button
  */
-function systemAlert (message, label, isConfirm, declineConfirmLabel, approveConfirmLabel, closeMessageLabel, hideOptionLabel, displayHideOption) {
+function systemAlert (message, label, isConfirm, declineConfirmLabel, approveConfirmLabel, closeMessageLabel, hideOptionLabel, displayHideOption, archiveMetadata, metadataWarningLabel) {
     declineConfirmLabel = declineConfirmLabel || (translateUI.t('dialog-cancel') || 'Cancel');
     approveConfirmLabel = approveConfirmLabel || (translateUI.t('dialog-confirm') || 'Confirm');
     closeMessageLabel = closeMessageLabel || (translateUI.t('dialog-ok') || 'Okay');
     hideOptionLabel = hideOptionLabel || (translateUI.t('dialog-hide') || "Don't ask again");
     displayHideOption = displayHideOption || false;
     label = label || (isConfirm ? 'Confirmation' : 'Message');
+    metadataWarningLabel = metadataWarningLabel || translateUI.t('dialog-metadata-warning') || 'Warning: above data can be spoofed!';
+    const metadataLabels = {
+        name: translateUI.t('dialog-metadata-name') || 'Name: ',
+        creator: translateUI.t('dialog-metadata-creator') || 'Creator: ',
+        publisher: translateUI.t('dialog-metadata-publisher') || 'Publisher: ',
+        scraper: translateUI.t('dialog-metadata-scraper') || 'Scraper: '
+    }
```

Comment:
Similarly, this code really doesn't belong here. In general, avoid mixing the code from a new feature into existing generic functions, otherwise things get confusing! It's a good principle of functional programming (although most of KJS was written with an object-oriented programming style...).